app/src/main/java/com/halim/adam/cashmaster/AddNewIncome.java

```diff
@@ -4,17 +4,54 @@
 import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
+import android.widget.EditText;
+
+import com.halim.adam.cashmaster.Objects.Budget;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
 
 public class AddNewIncome extends Activity {
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_add_new_income);
     }
 
-    public void MoveToTransactionHistoryPage(View view){
+    public void MoveToTransactionHistoryPage(View view) throws ParseException {
         Intent intent = new Intent(this, TransactionHistory.class);
         startActivity(intent);
+
+        EditText inputName = findViewById(R.id.inputName);
+        EditText inputAmount = findViewById(R.id.inputAmount);
+        EditText inputDate = findViewById(R.id.inputDate);
+
+        DatabaseHelper dbHelper = new DatabaseHelper(this);
+
+        if (!inputDate.getText().toString().matches("")){
+            Date date = DATE_FORMAT.parse(inputDate.getText().toString());
```

Comment:
For this part i think the parse will fail if the user input a random string into the inputDate, some kind of validation is probably needed (only contain numbers, format correct, etc)

Reply:
Hi @dhyatmikas, this pull request is closed already, please comment on the final phase pull request here

https://github.com/SeiryuZ/CashMaster/pull/2